src/lib/configs/project/spell.ts

```diff
@@ -0,0 +1,294 @@
+        nullable: true,
+        minimum: 0,
+        maximum: U32_MAX,
+      },
+      endDelaySec: {
+        type: "number",
+        description:
+          "How long to wait before the last execution in seconds. If this property or `endTimestamp` not specified, periodic execution will never end. WARNING! Currently your computer's clock is used to determine a final timestamp that is sent to the server. If it is in the past at the moment of spell creation - the spell will never be executed. This property overrides `endTimestamp` if both are specified",
+        nullable: true,
+        minimum: 0,
+        maximum: U32_MAX,
+      },
+    },
+    required: [],
+  },
+} as const;
+
+export const overridableSpellProperties = {
+  ...spellProperties,
+  aquaFilePath: {
+    ...spellProperties.aquaFilePath,
+    nullable: true,
+  },
+  function: {
+    ...spellProperties.function,
+    nullable: true,
+  },
+} as const;
+
+const configSchemaV0: JSONSchemaType<ConfigV0> = {
+  type: "object",
+  $id: `${TOP_LEVEL_SCHEMA_ID}/${SPELL_CONFIG_FILE_NAME}`,
+  title: SPELL_CONFIG_FILE_NAME,
+  description: `Defines a spell. You can use \`fluence spell new\` command to generate a template for new spell`,
+  properties: spellProperties,
+  required: ["version", "function", "aquaFilePath"],
+};
+
+const migrations: Migrations<Config> = [];
+type Config = ConfigV0;
+type LatestConfig = ConfigV0;
+export type SpellConfig = InitializedConfig<LatestConfig>;
+export type SpellConfigReadonly = InitializedReadonlyConfig<LatestConfig>;
+
+const getDateSec = (date: Date) => Math.round(date.getTime() / 1000);
+
+export const resolveStartSec = ({ clock }: LatestConfig): number => {
+  if (clock === undefined) {
+    return 0;
+  }
+
+  if (clock.startDelaySec !== undefined) {
+    return getDateSec(new Date()) + clock.startDelaySec;
+  }
+
+  return clock.startTimestamp === undefined
+    ? 1
+    : getDateSec(new Date(clock.startTimestamp));
+};
+
+export const resolveEndSec = ({ clock }: LatestConfig): number => {
+  if (clock === undefined) {
+    return 0;
+  }
+
+  if (clock.endDelaySec !== undefined) {
+    return getDateSec(new Date()) + clock.endDelaySec;
+  }
+
+  return clock.endTimestamp === undefined
+    ? 0
+    : getDateSec(new Date(clock.endTimestamp));
+};
+
+const validate: ConfigValidateFunction<LatestConfig> = async (
+  config,
+  configPath
+) =>
+  validateBatch(
+    config.clock?.startTimestamp !== undefined &&
+      config.clock?.startDelaySec !== undefined
+      ? `You can't specify both 'startTimestamp' and 'startDelaySec' properties`
+      : true,
+
+    config.clock?.endTimestamp !== undefined &&
+      config.clock?.endDelaySec !== undefined
+      ? `You can't specify both 'endTimestamp' and 'endDelaySec' properties`
```

Comment:
Inconsistent with the doc above. The same with `startTimestamp`.
> This property overrides `endTimestamp` if both are specified",

Reply:
Oh damn thank you for noticing it. I changed it and totally forgot about it. I personally think it would be better to have an error so it's more explicit what is happening